thejamesgore/Foxtons-Dashboard
# feat: Add neccessary interface components and TypeScript interfaces

## Description 🚀

This pull request adds the necessary interface for the entire project and includes the following changes:

- Added frontend files and components to support the necessary interface.
- Updated the project's README with instructions on how to run the frontend app.

## Changes Made 👉

- Created new directories and files for the necessary interface components under `src/components`.
- Added new React components for the necessary interface, including forms, buttons, and tables.
- Updated existing React components to use the new interface components.
- Added TypeScript interfaces for `agent.d.ts`, `common.d.ts`, `google.d.ts`, `home.d.ts`, `property.ts`, and `theme.d.ts`.


## Reason for Changes 🔍

By adding the necessary interface components to our project, we can improve the user experience and provide a more user-friendly interface for managing properties and tenants. Adding TypeScript interfaces for the necessary files also improves the maintainability and robustness of our codebase.

## How to Test

To test these changes, you can follow these steps:

- Clone this branch to your local machine.
- Install the necessary dependencies with `npm install`.
- Start the application with `npm start`.
- Verify that the application starts without any errors and that you can access the landlord dashboard pages by navigating to `http://localhost:3000/`.
- Test the new interface components by interacting with the forms, buttons, and tables.

## Checklist


## Related Issues

None.

## Other Notes

None.
